src/main/java/practice/StreamPractice.java

```diff
@@ -23,10 +38,17 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        OptionalDouble averageDouble = IntStream.range(0, numbers.size())
+                .map(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .boxed()
+                .filter(i -> i % 2 != 0)
+                .mapToDouble(i -> i)
+                .average();
+        return averageDouble.orElseThrow(NoSuchElementException::new);
```

Comment:
Why do we need boxed() and  .mapToDouble(i -> i) here? 
```suggestion
        return IntStream.range(0, numbers.size())
                .map(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
                .boxed()
                .filter(i -> i % 2 != 0)
                .mapToDouble(i -> i)
                .average()
                .orElseThrow();
```

lets just return lavue from stream